docs/website/docs/general-usage/credentials/configuration.md

```diff
@@ -0,0 +1,422 @@
+---
+title: Secrets and Configs
+description: Overview secrets and configs
+keywords: [credentials, secrets.toml, secrets, config, configuration, environment
+      variables]
+---
+
+# Secrets and Configs
+
+## General Usage and an Example
+
+This is what we had in mind, when designing:
+
+1. Adding configuration and secrets to sources and resources should be no-effort.
+2. You can reconfigure the pipeline for production after it is deployed. Deployed and local code should
+   be identical.
+3. You can always pass configuration values explicitly and override any default behavior (ie. naming of the configuration keys).
+
+
+The source below reads selected tabs from Google Sheets and demonstrates a few options you have to pass config and secret values:
+
+```python
+@dlt.source
+def google_sheets(
+    spreadsheet_id,
+    tab_names=dlt.config.value,
+    credentials=dlt.secrets.value,
+    only_strings=False
+):
+    sheets = build('sheets', 'v4', credentials=Services.from_json(credentials))
+    tabs = []
+    for tab_name in tab_names:
+        data = sheets.get(spreadsheet_id, tab_name).execute().values()
+        tabs.append(dlt.resource(data, name=tab_name))
+    return tabs
+```
```

Comment:
We need to continue explaining the reader the concept of configuration. Before we go into "wrong" please explain what happens in the example above:
- What is `tab_names`?
- What is `credentials`? How are they different from `tab_names`?
- What is `dlt.secrets.value`? Why do we need it?

Reply:
The answers are actually below in the "Right approach" section